File: apps/backend/src/main/java/com/digisphere/pontofacil/domain/company/Company.java
package com.digisphere.pontofacil.domain.company;

import com.digisphere.pontofacil.domain.adress.Address;
import com.digisphere.pontofacil.dtos.company.CompanyDTO;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity(name = "com_companies")
@AllArgsConstructor
@NoArgsConstructor
public class Company {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;
  private String name;
  private String image;

  @Embedded
  private Address address;

  public CompanyDTO toDto() {
    return new CompanyDTO(
      id,
      name,
      image,
      address
    );
  }
}

File: apps/backend/src/main/kotlin/com/digisphere/pontofacil/company/domain/Company.kt
package com.digisphere.pontofacil.company.domain

import com.digisphere.pontofacil.base.BaseEntity
import jakarta.persistence.Embedded
import jakarta.persistence.Entity
import jakarta.persistence.Table

@Entity(name = "com_companies")
@Table(name = "com_companies")
class Company(
  name: String,
  image: String,
  address: Address
) : BaseEntity() {

  var name: String = name
    protected set

  var image: String = image
    protected set

  @Embedded
  var address: Address = address
    protected set
}
